server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceShardsAllocator.java

```diff
@@ -124,19 +124,31 @@ protected void processInput(DesiredBalanceInput desiredBalanceInput) {
                 long index = desiredBalanceInput.index();
                 logger.debug("Starting desired balance computation for [{}]", index);
 
-                recordTime(
-                    cumulativeComputationTime,
-                    () -> setCurrentDesiredBalance(
-                        desiredBalanceComputer.compute(
-                            getInitialDesiredBalance(),
-                            desiredBalanceInput,
-                            pendingDesiredBalanceMoves,
-                            this::isFresh
-                        )
-                    )
-                );
+                DesiredBalance desiredBalance;
+                final long started = threadPool.relativeTimeInMillis();
+                try {
+                    desiredBalance = desiredBalanceComputer.compute(
+                        getInitialDesiredBalance(),
+                        desiredBalanceInput,
+                        pendingDesiredBalanceMoves,
+                        this::isFresh
+                    );
+                    setCurrentDesiredBalance(desiredBalance);
+                } finally {
+                    final long finished = threadPool.relativeTimeInMillis();
+                    cumulativeComputationTime.inc(finished - started);
+                }
```

Comment:
NIT: I believe this was changed as `desiredBalance` reference is required outside of `recordTime` action.
I think it is still possible to reuse that with atomic reference or by replacing Runnable with Supplier to minimize the code.

Reply:
Actually it always sets `setCurrentDesiredBalance` so the reference is not even needed, it is okay to rely on `currentDesiredBalance` after.

Reply:
🤦🏼 pushed 47cb12a62429358c5f96a4100a48faf3aaca0d67.